doc/supervised.rst

```diff
@@ -0,0 +1,171 @@
+
+NCA
+---
+
+Neighborhood Components Analysis (:class:`NCA`) is a distance
+metric learning algorithm which aims to improve the accuracy of nearest
+neighbors classification compared to the standard Euclidean distance. The
+algorithm  directly  maximizes  a stochastic  variant  of  the leave-one-out
+k-nearest neighbors (KNN) score on the training set.  It can also learn a
+low-dimensional linear  embedding  of  data  that  can  be used for data
+visualization and fast classification.
+
+.. rubric:: Example Code:
+
+::
+
+    import numpy as np
+    from metric_learn import NCA
+    from sklearn.datasets import load_iris
+
+    iris_data = load_iris()
+    X = iris_data['data']
+    Y = iris_data['target']
+
+    nca = NCA(max_iter=1000, learning_rate=0.01)
+    nca.fit(X, Y)
+
+.. rubric:: References:
+
+.. [1] J. Goldberger, G. Hinton, S. Roweis, R. Salakhutdinov.
+"Neighbourhood Components Analysis". Advances in Neural Information
+Processing Systems. 17, 513-520, 2005.
+http://www.cs.nyu.edu/~roweis/papers/ncanips.pdf
+
+.. [2] Wikipedia entry on Neighborhood Components Analysis
+https://en.wikipedia.org/wiki/Neighbourhood_components_analysis
+
+LFDA
+----
+
+Local Fisher Discriminant Analysis (LFDA)
+
+:class:`LFDA` is a linear supervised dimensionality reduction
+method. It is particularly useful when dealing with multimodality, where one
+ore more classes consist of separate clusters in input space. The core
+optimization problem of LFDA is solved as a generalized eigenvalue problem.
+
+.. rubric:: Example Code:
+
+::
+
+    import numpy as np
+    from metric_learn import LFDA
+    from sklearn.datasets import load_iris
+
+    iris_data = load_iris()
+    X = iris_data['data']
+    Y = iris_data['target']
+
+    lfda = LFDA(k=2, dim=2)
+    lfda.fit(X, Y)
+
+.. rubric:: References:
+
+`Dimensionality Reduction of Multimodal Labeled Data by Local Fisher Discriminant Analysis <http://www.ms.k.u-tokyo.ac.jp/2007/LFDA.pdf>`_ Masashi Sugiyama.
+
+`Local Fisher Discriminant Analysis on Beer Style Clustering <https://gastrograph.com/resources/whitepapers/local-fisher-discriminant-analysis-on-beer-style-clustering.html#>`_ Yuan Tang.
+
+
+MLKR
+----
+
+:class:`MLKR` is an algorithm for supervised metric learning,
+which learns a distance function by directly minimising the leave-one-out
+regression error. This algorithm can also be viewed as a supervised variation
+of PCA and can be used for dimensionality reduction and high dimensional data
+visualization.
+
+.. rubric:: Example Code:
+
+::
+
+    from metric_learn import MLKR
+    from sklearn.datasets import load_iris
+
+    iris_data = load_iris()
+    X = iris_data['data']
+    Y = iris_data['target']
+
+    mlkr = MLKR()
+    mlkr.fit(X, Y)
+
+.. rubric:: References:
+
+`Information-theoretic Metric Learning <http://machinelearning.wustl.edu/mlpapers/paper_files/icml2007_DavisKJSD07.pdf>`_ Jason V. Davis, et al.
```

Comment:
this is not the right reference (this is for ITML)

Reply:
Thanks, didn't see that
Done